doc/contrib/pr.rst

```diff
@@ -48,6 +62,37 @@ Refer to the `Github
 documentation <https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`__
 for more explanation of the Git workflow described above.
 
+.. _contribpr-collab:
+
+Collaborating on a PR
+---------------------
+
+Especially for larger PRs, it may be a good idea to collaborate with others. There are various ways to do so,
+for instance to have a shared branch somewhere in a fork of the Arbor repo. Since we're using Github, you can
+also add commits to a PR opened by someone else. Since the correct procedure for that is 
+`Github specific <https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork>`_,
+here the steps for how to do this (with just ``git``).
+
+Alongside the instruction, an example situation will be given assuming you have the Arbor main
+repo setup as a remote called ``upsteam``, you want to commit to pull request number ``321``, opened by a 
+Github user called ``github_user_007``, who named the branch they're trying to merge ``my_special_branch``,
+in their fork of the Arbor repo called ```arbor-sim`` (which is the default when you fork on Github):
+
+1. ``git fetch $REMOTE pull/#PR/head:$REMOTE_BRANCH_NAME``
+
+   Example: ``git fetch upstream pull/321/head:my_special_branch``
+2. This should have made a local branch with the same name, tracking the PR-branch. Switch to the new local branch.
+
+   Example: ``git switch my_special_branch``
+3. Make commits to that local branch.
+4. Push to the PR-branch with the following incantation:
+   ``git push git@github.com:$USER/$REPO $LOCAL_BRANCH_NAME:$REMOTE_BRANCH_NAME``
+
+   Example: ``git push git@github.com:github_user_007/arbor-sim my_special_branch:my_special_branch``
```

Comment:
Should you push against the fork of the PR author here, not $USER?

Reply:
Yes, that can be clearer indeed.